### PR TITLE
Read stats annotations directly from pod info volume.

### PR DESCRIPTION
**Pull request by bianpengyuan**
This will help to clean up ISTIO_METAJSON env var (https://github.com/istio/istio/pull/32295), which passes duplicated information about pod annotations as pod-info downward API volume.

### CODE REVIEW

**Review by howardjohn — APPROVED** [review]

File: pkg/bootstrap/config.go [hunk 1]
```diff
@@ -180,6 +184,20 @@ var DefaultStatTags = []string{
...
+	if err != nil {
+		if os.IsNotExist(err) {
+			log.Debugf("failed to read pod annotations: %v", err)
+		} else {
+			log.Warnf("failed to read pod annotations: %v", err)
+		}
+	}
+
+	tagAnno := annotations[extraTagsAnnotation]
+	prefixAnno := annotations[annotation.SidecarStatsInclusionRegexps.Name]
```
Comment (kyessenov) [inline pkg/bootstrap/config.go:197]: bug: should be `InclusionPrefixes` here

**Review by kyessenov — APPROVED** [review]
This is an improvement.

File: pilot/pkg/model/context.go [hunk 2]
```diff
@@ -443,11 +443,6 @@ type BootstrapNodeMetadata struct {
 	// of the workload instance (ex: k8s deployment for a k8s pod).
 	Owner string `json:"OWNER,omitempty"`
 
-	StatsInclusionPrefixes string `json:"sidecar.istio.io/statsInclusionPrefixes,omitempty"`
-	StatsInclusionRegexps  string `json:"sidecar.istio.io/statsInclusionRegexps,omitempty"`
-	StatsInclusionSuffixes string `json:"sidecar.istio.io/statsInclusionSuffixes,omitempty"`
-	ExtraStatTags          string `json:"sidecar.istio.io/extraStatTags,omitempty"`
-
```
Comment (kyessenov) [inline pilot/pkg/model/context.go:450]: I think I would prefer that bootstrap node metadata had annotations and labels fields to satisfy the purity of the bootstrap derivation. The way it is now, the function reads files so it is hard to port this function into another place to compute.

Comment (bianpengyuan) [reply pilot/pkg/model/context.go:450]: We can do the same as labels: https://github.com/istio/istio/blob/3d7671c491d678aff701eb9a0cb48006494ca391/pkg/bootstrap/config.go#L481 and have a dedicated Annotations section.

**Comment by kyessenov** [comment]
Yeah that works.

**Comment by kyessenov** [comment]
BTW, do we know how this is supposed to work on VMs? Is pod info file available there and is it safe to remove environment variable for annotations? No need to answer if you have no answers :)

**Comment by bianpengyuan** [comment]
> BTW, do we know how this is supposed to work on VMs? Is pod info file available there and is it safe to remove environment variable for annotations? No need to answer if you have no answers :)

I suppose using stats matcher in ProxyConfig instead of annotations.

**Comment by bianpengyuan** [comment]
@kyessenov mind taking another look? A lot of changes were made since last approval.

**Review by kyessenov — APPROVED** [review]

**Pull request merged**